bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/profiles/RawRockerToDimmerProfile.java

```diff
@@ -0,0 +1,129 @@
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.IncreaseDecreaseType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.CommonTriggerEvents;
+import org.eclipse.smarthome.core.thing.profiles.ProfileCallback;
+import org.eclipse.smarthome.core.thing.profiles.ProfileContext;
+import org.eclipse.smarthome.core.thing.profiles.ProfileTypeUID;
+import org.eclipse.smarthome.core.thing.profiles.SystemProfiles;
+import org.eclipse.smarthome.core.thing.profiles.TriggerProfile;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+
+/**
+ * The {@link RawRockerToDimmerProfile} transforms rocker switch channel events into dimmer commands.
+ *
+ * @author Jan Kemmler - Initial contribution
+ */
+@NonNullByDefault
+public class RawRockerToDimmerProfile implements TriggerProfile {
+
+    private final ProfileCallback callback;
+
+    ProfileContext context;
+
+    @Nullable
+    private ScheduledFuture<?> dimmFuture;
+    @Nullable
+    private ScheduledFuture<?> timeoutFuture;
+
+    private long pressedTime = 0;
+
+    final class DimmerIncreaseDecreaseTask implements Runnable {
+        private ProfileCallback callback;
+        private Command command;
+
+        DimmerIncreaseDecreaseTask(ProfileCallback callback, Command command) {
+            this.callback = callback;
+            this.command = command;
+        }
+
+        @Override
+        public void run() {
+            callback.sendCommand(command);
+        }
+    };
+
+    @SuppressWarnings("null")
+    RawRockerToDimmerProfile(ProfileCallback callback, ProfileContext context) {
+        this.callback = callback;
+        this.context = context;
+    }
+
+    @Override
+    public ProfileTypeUID getProfileTypeUID() {
+        return SystemProfiles.RAWROCKER_TO_DIMMER;
+    }
+
+    @Override
+    public void onStateUpdateFromItem(State state) {
+
+    }
+
+    @Override
+    public void onTriggerFromHandler(String event) {
+        if (CommonTriggerEvents.DIR1_PRESSED.equals(event)) {
+            buttonPressed(IncreaseDecreaseType.INCREASE);
+        } else if (CommonTriggerEvents.DIR1_RELEASED.equals(event)) {
+            buttonReleased(OnOffType.ON);
+        } else if (CommonTriggerEvents.DIR2_PRESSED.equals(event)) {
+            buttonPressed(IncreaseDecreaseType.DECREASE);
+        } else if (CommonTriggerEvents.DIR2_RELEASED.equals(event)) {
+            buttonReleased(OnOffType.OFF);
+        }
+    }
+
+    private void buttonPressed(Command commandToSend) {
+        if (null != timeoutFuture) {
+            timeoutFuture.cancel(false);
+        }
+        if (null != dimmFuture) {
+            dimmFuture.cancel(false);
+        }
+
+        dimmFuture = context.getExecutorService().scheduleWithFixedDelay(
```

Comment:
When you write 

```java 
dimmFuture = context.getExecutorService().scheduleWithFixedDelay(() -> callback.sendCommand(commandToSend), 550, 200, TimeUnit.MILLISECONDS);
```

instead, you will save 15 lines of superfluous code above.

Reply:
Ah okay, that's what lambdas look like in Java. Thanks, this is much nicer!